src/backend/distributed/connection/connection_configuration.c

```diff
@@ -260,15 +260,30 @@ GetConnParams(ConnectionHashKey *key, char ***keywords, char ***values,
 	/* auth keywords will begin after global and runtime ones are appended */
 	Index authParamsIdx = ConnParams.size + lengthof(runtimeKeywords);
 
+	int paramIndex = 0;
+	int runtimeParamIndex = 0;
+
 	pg_ltoa(key->port, nodePortString); /* populate node port string with port */
 
 	/* first step: copy global parameters to beginning of array */
-	memcpy(connKeywords, ConnParams.keywords, ConnParams.size * sizeof(char *));
-	memcpy(connValues, ConnParams.values, ConnParams.size * sizeof(char *));
+	for (paramIndex = 0; paramIndex < ConnParams.size; paramIndex++)
+	{
+		/* copy the keyword&value pointers to the new array */
+		connKeywords[paramIndex] = ConnParams.keywords[paramIndex];
+		connValues[paramIndex] = ConnParams.values[paramIndex];
+	}
 
 	/* second step: begin at end of global params and copy runtime ones */
-	memcpy(&connKeywords[ConnParams.size], runtimeKeywords, sizeof(runtimeKeywords));
-	memcpy(&connValues[ConnParams.size], runtimeValues, sizeof(runtimeValues));
+    for (runtimeParamIndex = 0;
+         runtimeParamIndex < lengthof(runtimeKeywords);
+         runtimeParamIndex++)
+    {
+        /* copy the keyword&value pointers to the new array */
+        connKeywords[ConnParams.size + runtimeParamIndex] =
```

Comment:
In the spirit of these changes, should we add some ConnParams.size < ConnParams.maxSize check somewhere?